frontend/packages/console-plugin-sdk/src/registry.ts

```diff
@@ -55,9 +56,19 @@ export class ExtensionRegistry {
       );
   }
 
+  public getBlockerFlags(typeGuards: ExtensionTypeGuard<ExtensionWithFlags>[]) {
+    return _.flatMap(typeGuards.map((tg) => this.extensions.filter(tg)))
+      .filter((e) => e.properties.blocker)
+      .reduce(
+        (blockerFlags, e) => _.uniq([...blockerFlags, ..._.castArray(e.properties.blocker)]),
+        [] as string[],
+      );
+  }
+
   public isExtensionInUse(e: ExtensionWithFlags, flags: FlagsObject) {
+    const blockerFlags = e.properties.blocker ? _.castArray(e.properties.blocker) : [];
     const requiredFlags = e.properties.required ? _.castArray(e.properties.required) : [];
-    return _.every(requiredFlags, (f) => flags[f]);
+    return _.every(requiredFlags, (f) => flags[f]) && _.every(blockerFlags, (f) => !flags[f]);
```

Comment:
```suggestion
    return _.every(requiredFlags, (f) => flags[f]) && _.every(disallowedFlags, (f) => !flags[f]);
```

Reply:
Also `!flags[f]` is truthy even if the flag has `undefined` value, so we should improve the current code:

```ts
return _.every(requiredFlags, (f) => flags[f] === true) && _.every(disallowedFlags, (f) => flags[f] === false);
```